The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9344155399/job/25714849893) [(plain)](https://github.com/rust-lang/rust/commit/30ef37b0ed24391ff17048cf3641fd9965af22fd/checks/25714849893/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
downloading https://static.rust-lang.org/dist/2024-05-26/cargo-beta-x86_64-unknown-linux-gnu.tar.xz
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1204, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1189, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1154, in bootstrap
    build.download_toolchain()
  File "/checkout/src/bootstrap/bootstrap.py", line 604, in download_toolchain
    shutil.rmtree(self.bootstrap_out())
  File "/usr/lib/python3.10/shutil.py", line 715, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib/python3.10/shutil.py", line 713, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/checkout/obj/build/bootstrap'
Command failed. Attempt 2/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1204, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1189, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1154, in bootstrap
    build.download_toolchain()
  File "/checkout/src/bootstrap/bootstrap.py", line 604, in download_toolchain
    shutil.rmtree(self.bootstrap_out())
  File "/usr/lib/python3.10/shutil.py", line 715, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib/python3.10/shutil.py", line 713, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/checkout/obj/build/bootstrap'
Command failed. Attempt 3/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1204, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1189, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1154, in bootstrap
    build.download_toolchain()
  File "/checkout/src/bootstrap/bootstrap.py", line 604, in download_toolchain
    shutil.rmtree(self.bootstrap_out())
  File "/usr/lib/python3.10/shutil.py", line 715, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib/python3.10/shutil.py", line 713, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/checkout/obj/build/bootstrap'
Command failed. Attempt 4/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1204, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1189, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1154, in bootstrap
    build.download_toolchain()
  File "/checkout/src/bootstrap/bootstrap.py", line 604, in download_toolchain
    shutil.rmtree(self.bootstrap_out())
  File "/usr/lib/python3.10/shutil.py", line 715, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib/python3.10/shutil.py", line 713, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/checkout/obj/build/bootstrap'
Command failed. Attempt 5/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1204, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1189, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1154, in bootstrap
    build.download_toolchain()
  File "/checkout/src/bootstrap/bootstrap.py", line 604, in download_toolchain
    shutil.rmtree(self.bootstrap_out())
  File "/usr/lib/python3.10/shutil.py", line 715, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib/python3.10/shutil.py", line 713, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/checkout/obj/build/bootstrap'
The command has failed after 5 attempts.
  local time: Mon Jun  3 04:15:14 UTC 2024
  network time: Mon, 03 Jun 2024 04:15:14 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/src/bootstrap/bootstrap.py
+++ b/src/bootstrap/bootstrap.py
@@ -599,6 +599,12 @@ class RustBuild(object):
                 print('Choosing a pool size of', pool_size, 'for the unpacking of the tarballs')
             p = Pool(pool_size)
             try:
+                # FIXME: A cheap workaround for https://github.com/rust-lang/rust/issues/125578,
+                # remove this once the issue is closed.
+                bootstrap_out = self.bootstrap_out()
+                if os.path.exists(bootstrap_out):
+                    shutil.rmtree(bootstrap_out)
+
                 p.map(unpack_component, tarballs_download_info)
             finally:
                 p.close()
@@ -864,6 +870,16 @@ class RustBuild(object):
             return line[start + 1:end]
         return None
 
+    def bootstrap_out(self):
+        """Return the path of the bootstrap build artifacts
+
+        >>> rb = RustBuild()
+        >>> rb.build_dir = "build"
+        >>> rb.bootstrap_binary() == os.path.join("build", "bootstrap")
+        True
+        """
+        return os.path.join(self.build_dir, "bootstrap")
+
     def bootstrap_binary(self):
         """Return the path of the bootstrap binary
 
@@ -873,7 +889,7 @@ class RustBuild(object):
         ... "debug", "bootstrap")
         True
         """
-        return os.path.join(self.build_dir, "bootstrap", "debug", "bootstrap")
+        return os.path.join(self.bootstrap_out(), "debug", "bootstrap")
 
     def build_bootstrap(self):
         """Build bootstrap"""

